src/components/App.js

```diff
@@ -150,6 +161,28 @@ class App extends Component {
     setActiveSearch("symbol");
   }
 
+  toggleGoToLineModal(_, e: SyntheticEvent) {
+    const {
+      selectedSource,
+      activeSearch,
+      closeActiveSearch,
+      setActiveSearch
+    } = this.props;
+
+    e.preventDefault();
+    e.stopPropagation();
+
+    if (!selectedSource) {
+      return;
+    }
+
+    if (activeSearch == "line") {
+      return closeActiveSearch();
+    }
+
+    setActiveSearch("line");
```

Comment:
we'll likely need to update the flow definition for the possible search types in `reducers/ui.js`

Reply:
do you mean [here](https://github.com/devtools-html/debugger.html/blob/master/src/reducers/ui.js#L21)? I am surprised flow did not catch that for me in a warning... I guess I don't really know how flow works 😃